compiler/compiler-core.h

```diff
@@ -38,6 +41,7 @@ class CompilerCore {
   std::vector<std::string> kphp_runtime_opts;
   bool is_untyped_rpc_tl_used{false};
   function_palette::Palette function_palette;
+  std::map<std::string, const ClassMemberInstanceField *> job_workers_shared_memory_pieces;
```

Comment:
This is not to fixed in this PR, just a bit to discuss. This field seems overcomplicated for me.
I suppose, that adding `bool has_job_shared_memory_piece` into `ClassData`, setting it to true once (where you  add a value to this map) and a getter method in `ClassData` that just uses full lookup through instance fields, would be enough and more clear that storing a global map here

Reply:
Hmm. Probably yes, there won't be too many classes containing `KphpJobWorkerSharedMemoryPiece` field, and so we simply can perform full lookup twice for them. I didn't think about that. I agree it would be more clear

Reply:
done